### PR TITLE
Analyzed the tests for the class TaskExecutor

### LINKED ISSUE #11: Analyze the tests
The tests have to analyzed in order to answer the following questions:

* What are the tests?
* Do the tests cover the requirements?
* Does your refactoring introduce anything that requires more tests?

### PR DESCRIPTION
**Pull request by francislaus**
This answered the following questions:

* What are the tests?
* Do the tests cover the requirements?
* Does your refactoring introduce anything that requires more tests?

This should close #11

### CODE REVIEW

**Review by apeinot — CHANGES_REQUESTED** [review]
In the last paragraph, the two first sentences are contradictory. To be fix

**Review by Samuelhertzberg — CHANGES_REQUESTED** [review]
"Therefore, we do not need to add extra test cases." To be removed
"6/7" => "six out of seven"

**Comment by francislaus** [comment]
Made changes accordingly to the comments. I was not quite sure about the comments by @apeinot, I hope I did what you requested

**Review by emilelm96 — APPROVED** [review]

**Review by apeinot — APPROVED** [review]

**Review by Samuelhertzberg — APPROVED** [review]

File: report.md [hunk 1]
```diff
@@ -55,8 +55,10 @@ The following table will cover all requirements as they are described in previou
 | Throws an execption if invalid parameters given | There is no test case, which tests if an execption is thrown if invalid parameters are given. We will write such a test case.|
 | Ensures the right sequence of operations | The different flow of operations is tested by six out of seven test cases, only  `testSourceVertexDataFetching` tests one operation. Other test cases cover cases with multiple incoming and outgoing edges for one vertex.|
 
+## New test cases relating to refactored code
 
-
+One new test was added. The input data to the getTaskExecutor() function is set to entirely `null`. This should result in a nullPointerException, which the new test catch. If no nullPointerException is thrown, something is very wrong and the new test fails.
```
Comment (Jonagoo) [inline report.md]: catches*

**Review by emilelm96 — COMMENTED** [review]
NullPointerException should be spelled with a big N.

**Review by Jonagoo — CHANGES_REQUESTED** [review]

File: report.md (hunk 1 above)
Comment (francislaus) [reply report.md]: *NullPointerException

**Review by emilelm96 — APPROVED** [review]

**Review by Jonagoo — APPROVED** [review]

**Pull request merged**